lib/internal/bootstrap_node.js

```diff
@@ -319,10 +319,10 @@
   }
 
   function setupInspector(originalConsole, wrappedConsole, Module) {
-    const { addCommandLineAPI, consoleCall } = process.binding('inspector');
-    if (!consoleCall) {
+    if (!process.config.variables.v8_enable_inspector) {
```

Comment:
Depending on `process.config` is dangerous because there are userland modules that completely replace it. During bootstrap it *should* be safe but problems still could come up.

Reply:
If it ever becomes an issue, we could cache it before user code runs.

Reply:
Or use `process.binding('config')` as a stable alternative.

Reply:
@jasnell @cjihrig Thanks, I had not thought about that being an issue and I'm trying to understand how this might happen. 

Using `process.binding('config')` will give access to the builtin module `config`, but as far as I can tell the information from config.gypi is provided as a native module (via node_js2c). The process object is then configured with that information in [setupConfig](https://github.com/nodejs/node/blob/882e811c88949a87ca2c0a7674a84c1932a7850d/lib/internal/process.js#L118-L136) which is called by node_bootstrap.js [start](https://github.com/nodejs/node/blob/882e811c88949a87ca2c0a7674a84c1932a7850d/lib/internal/bootstrap_node.js#L30-L31) function. 

`setupConfig` deletes the config from the _source and then sets these properties on the process object. Since it deletes the _source this function cannot be called multiple times, for example if it did not we could have done the same thing again.

What I'm having some difficulties understand is that how could a userland module be able to replace this at this stage. Would someone be able to shed some light on this?

Reply:
Userland code *might not* be able to replace `process.config` by this stage, I haven't tested it. What I do know is that we've had issues depending on `process.config` in later stages elsewhere in the code and we if we are going to use it here then we need to be certain that it won't be a problem. If there is no userland code that can be run *before* bootstrap gets to this point, then it *should* be ok.

Reply:
This is still very early in the bootstrap process and I do not think it is possible to change it at that stage. To be on the safe side we could just add a test for it though.

Reply:
@jasnell @BridgeAR Sounds good, I'll add a test for this. Thanks